src/homebridge-ui/public/index.html

```diff
@@ -0,0 +1,77 @@
+<style>
+    .header-divider {
+        margin-top: .25rem;
+    }
+
+    #authButton:disabled {
+        cursor: not-allowed;
+        pointer-events: all !important;
+    }
+
+    #authUrlInput:disabled {
+        cursor: not-allowed;
+        pointer-events: all !important;
+    }
+</style>
+
+<div>
+    <h5>Getting Started</h5>
+    <hr class="header-divider"/>
+    <div class="card card-body">
+        <ol>
+            <li>
+                Follow the steps on the Google SDM
+                <a target="_blank" href="https://developers.google.com/nest/device-access/get-started">Getting Started</a>
+                page to obtain a client ID, client secret, SDM project ID, & PubSub Subscription. Once these values are entered into the config form below,
+                the 'Initiate Google Login' button will be enabled.
+            </li>
+            <li>
+                After initiating Google login, enable all permissions and continue through any warnings about an
+                unverified app. Once you get through all the prompts, you will be redirected to <code>google.com</code>.
+            </li>
+            <li>
+                Copy the <code>google.com</code> URL (it will contain a <code>code=</code> segment) into the 'Auth URL' text box below.
+            </li>
+            <li>
+                A refresh token will then be generated and auto-populated into the config form below.
+            </li>
+        </ol>
+    </div>
+
+    <h5 style="margin-top: 40px">Generate Refresh Token</h5>
+    <hr class="header-divider"/>
+
+    <p>
+        <small><em>Enter the client ID, client secret, SDM project ID, & PubSub subscription config values to enable this form.</em></small>
+    </p>
+    <div class="card card-body">
+        <button id="authButton" type="button" class="btn btn-primary" style="margin-top: 10px">
+            Initiate Google Login
+        </button>
+
+        <div id="auth-form-group" style="margin: 10px 0 10px 3px">
+            <label for="authUrlInput" class="form-label">Auth URL</label>
+            <div style="display: flex">
+                <input type="url" class="form-control" id="authUrlInput" placeholder="Paste auth URL here" style="width: 80%; margin-right: 15px">
+                <div id="tokenLoadingSpinner" class="spinner-border text-primary" role="status" style="visibility: hidden">
+                    <span class="sr-only">Loading...</span>
+                </div>
+            </div>
+        </div>
+    </div>
+
+    <h5 style="margin-top: 40px">Config</h5>
+    <hr class="header-divider"/>
+</div>
+
+<script src="https://cdn.jsdelivr.net/npm/lodash@4.17.21/lodash.min.js"></script>
+<script>
+    // Create a script tag to load the JS onto the page
+    const script = document.createElement('script')
```

Comment:
Why do this instead of simply referencing the script directly?  Something about the cache? Is reloading the page not sufficient?

Reply:
This ensures that the latest `nest-sdm-ui.js` is being used. If the script's `src` doesn't change between releases, the browser will sometimes use a cached version, even through page reloads.

This solution was inspired by the [Ring plugin](https://github.com/dgreif/ring/blob/main/packages/homebridge-ring/homebridge-ui/public/index.html)

Reply:
The alternative is to have all of the code within the HTML file (e.g. [homebridge-meross](https://github.com/bwp91/homebridge-meross/blob/latest/lib/homebridge-ui/public/index.html)). I generally like separating the code from the UI, but if you prefer it all-in-one, I can definitely make that change.